tetrahedron/src/main/java/com/epam/jwd/register/RegisterTetrahedron.java

```diff
@@ -0,0 +1,65 @@
+package com.epam.jwd.register;
+
+import com.epam.jwd.action.SurfaceAreaTetrahedron;
+import com.epam.jwd.action.VolumeTetrahedron;
+import com.epam.jwd.entity.ShapeTetrahedron;
+import com.epam.jwd.entity.Tetrahedron;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class RegisterTetrahedron implements Observer<Tetrahedron>{
+
+    private static final Logger LOG = LogManager.getLogger(RegisterTetrahedron.class.getName());
+
+    private static RegisterTetrahedron registerTetrahedron;
+    private Map<Integer, List<Double>> register;
+
+
+    private RegisterTetrahedron() {
+        this.register = new HashMap<>();
+    }
+
+    public static RegisterTetrahedron getRegisterTetrahedron(){
+        if (registerTetrahedron == null) {
+            registerTetrahedron = new RegisterTetrahedron();
+        }
+        return registerTetrahedron;
+    }
+
+    @Override
+    public void handleEvent(Tetrahedron tetrahedron) {
+        register.replace(tetrahedron.getTetrahedronId(), calculateParameters(tetrahedron));
+        LOG.info("Changed tetrahedron's parameters with id={} on {}", tetrahedron.getTetrahedronId(), calculateParameters(tetrahedron));
+    }
+
+    public void addTetrahedron(Tetrahedron tetrahedron) {
+        register.put(tetrahedron.getTetrahedronId(), calculateParameters(tetrahedron));
+        LOG.info("Add tetrahedron to register {}", tetrahedron);
+
+    }
+
+    private List<Double> calculateParameters(ShapeTetrahedron tetrahedron) {
+
+        List<Double> parameters = new ArrayList<>();
+        SurfaceAreaTetrahedron surfaceAreaTetrahedron = new SurfaceAreaTetrahedron();
+        Double surfaceArea = surfaceAreaTetrahedron.calculateSurfaceAreaTetrahedron((Tetrahedron)tetrahedron);
```

Comment:
broher, i think make acion classes singletone and in registrator recalculation method only call action methods